plugins/pci-psp/fu-plugin-pci-psp.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (C) 2022 Advanced Micro Devices Inc.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+struct FuPluginData {
+	gchar *sysfs_path;
+};
+
+static void
+fu_plugin_pci_psp_init(FuPlugin *plugin)
+{
+	fu_plugin_alloc_data(plugin, sizeof(FuPluginData));
+	fu_plugin_add_udev_subsystem(plugin, "pci");
+}
+
+static void
+fu_plugin_pci_psp_destroy(FuPlugin *plugin)
+{
+	FuPluginData *priv = fu_plugin_get_data(plugin);
+	g_free(priv->sysfs_path);
+}
+
+static gboolean
+fu_plugin_pci_psp_backend_device_added(FuPlugin *plugin, FuDevice *device, GError **error)
+{
+	FuPluginData *priv = fu_plugin_get_data(plugin);
+
+	/* interesting device? */
+	if (!FU_IS_UDEV_DEVICE(device))
+		return TRUE;
+	if (g_strcmp0(fu_udev_device_get_subsystem(FU_UDEV_DEVICE(device)), "pci") != 0)
+		return TRUE;
+
+	priv->sysfs_path = g_strdup(fu_udev_device_get_sysfs_path(FU_UDEV_DEVICE(device)));
```

Comment:
I wonder if you'd rather store a ref to the FuDevice instead of just the sysfs_path?